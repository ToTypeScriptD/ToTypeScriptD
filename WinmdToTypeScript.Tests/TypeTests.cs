using ApprovalTests;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WinmdToTypeScript.Tests
{

    public class TypeTests : TestBase
    {
        [Test]
        public void EnumType()
        {
            var result = GetNativeType("SampleEnum").ToTypeScript();
            Approvals.Verify(result);
        }

        [Test]
        public void EnumTypeWithExplicitValues()
        {
            var result = GetNativeType("SampleEnumNumbered").ToTypeScript();
            Approvals.Verify(result);
        }

        [Test]
        public void ClassWithEventHandler()
        {
            var result = GetNativeType("ClassWithEventHandler").ToTypeScript();
            Approvals.Verify(result);
        }

        [Test]
        public void FullSampleAssembly()
        {
            var result = WinmdToTypeScript.Render.FullAssembly(base.NativeComponentPath);
            Approvals.Verify(result);
        }

        [Test]
        public void WindowsStorageClass()
        {
            var result = GetWinNativeType("Windows.Storage.StorageFile").ToTypeScript();
            Approvals.Verify(result);
        }
        [Test]
        public void WindowsSystemUserProfileGlobalizationPreferencesClass()
        {
            var result = GetWinNativeType("Windows.System.UserProfile.GlobalizationPreferences").ToTypeScript();
            Approvals.Verify(result);
        }
        
        //[Test]
        //public void FullWindowsAssembly()
        //{
        //    var file = @"C:\Windows\System32\WinMetadata\Windows.Foundation.winmd";
        //    var result = WinmdToTypeScript.Render.FullAssembly(file);
        //    Approvals.Verify(result);
        //}
    }
}
