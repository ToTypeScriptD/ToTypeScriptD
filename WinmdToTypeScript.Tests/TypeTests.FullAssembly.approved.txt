declare module WinmdToTypeScript.Native {
    interface SomethingHappenedEventHandler{
    }
}
declare module WinmdToTypeScript.Native {
    export class ClassWithEventHandler{
        addEventListener(type: string, listener: EventListener): void;
        removeEventListener(type: string, listener: EventListener): void;
        onsomethinghappened(ev: any);
        constructor();
        doSomething();
    }
}
declare module WinmdToTypeScript.Native {
    export class Class1{
        constructor();
    }
}
declare module WinmdToTypeScript.Native {
    enum SampleEnum{
        A = 0,
        B = 1,
        C = 2,
        D = 3
    }
}
declare module WinmdToTypeScript.Native {
    enum SampleEnumNumbered{
        A = 1,
        B = 10,
        C = 100,
        D = 99
    }
}
